templates/frontend/pages/article.tpl

```diff
@@ -23,18 +31,51 @@
 
 	{if $galley}
 		<h1 class="page_title">{$article->getLocalizedTitle()|escape}</h1>
-
-		{translate key="article.view.interstitial" galleyUrl=$fileUrl}
-		<ul class="galleys_links">
-			{foreach from=$galley->getLatestGalleyFiles() item=galleyFile}
-				<li>
-                    <a class="obj_galley_link" href="{url op="download" path=$article->getBestArticleId($currentJournal)|to_array:$galley->getBestGalleyId($currentJournal):$galleyFile->getFileId() escape=false}">{$galleyFile->getLocalizedName()|escape}</a>
-                </li>
-			{/foreach}
-		</ul>
+		{if $galley->hasFilesPerSubmissionRevision($submissionRevision)}
+			{translate key="article.view.interstitial" galleyUrl=$fileUrl}
+			<form id='fileDownloadForm'>
+				<ul class="galleys_links">
+					{foreach from=$galley->getLatestGalleyFiles() item=galleyFile}
+						<li>
+							{assign var=downloadLink value="file"|to_array:$article->getBestArticleId($currentJournal):$galley->getBestGalleyId($currentJournal):$submissionRevision:$galleyFile->getFileId()}
+							<a class="obj_galley_link" href="{url op="download" path=$downloadLink escape=false}">{$galleyFile->getLocalizedName()|escape}</a>
+							{assign var=otherRevisions value=$galley->getOtherRevisions($galleyFile->getFileId(), null, $galleyFile->getSubmissionId(), $submissionRevision)}
+							{if $otherRevisions && !$galleyFile->getHideFileRevisions()}
+							<div class="revisions">
+								<span>{translate key="article.revisions"}:</span>
+								{fbvElement type="select" from=$otherRevisions class="revisions" id="revisions" name="revisions" translate=false}
+								<a href="{url op="download" path=$downloadLink escape=false}">{translate key="common.download"}</a>
+							</div>
+							{/if}
+						</li>
+					{/foreach}
+				</ul>
+			</form>
+		{else}
+			{translate key="submission.noGalleyFiles"}
+		{/if}
 	{else}
 		{* Show article overview *}
 		{include file="frontend/objects/article_details.tpl"}
+		
+		{if $isPreviousRevision}
```

Comment:
1) It was intentional to display just the link to the newest version when you are on an older revision.
2) You actually should see the links to all older versions when you are on the newest version (unless the "Hide links to previous versions of this submission" checkbox is activated)